WooCommerce/Classes/ViewRelated/Editor/FormatBar/FormatBarItemViewProperties.swift

```diff
@@ -0,0 +1,114 @@
+import UIKit
+import Gridicons
+import Aztec
+
+protocol FormatBarItemViewProperties {
+    var iconImage: UIImage { get }
+    var accessibilityLabel: String { get }
+}
+
+// MARK: - FormattingIdentifier
+//
+extension FormattingIdentifier: FormatBarItemViewProperties {
+    var iconImage: UIImage {
+        switch self {
+        case .media:
+            return Gridicon.iconOfType(.addOutline)
+        case .p:
+            return Gridicon.iconOfType(.heading)
+        case .bold:
+            return Gridicon.iconOfType(.bold)
+        case .italic:
+            return Gridicon.iconOfType(.italic)
+        case .underline:
+            return Gridicon.iconOfType(.underline)
+        case .strikethrough:
+            return Gridicon.iconOfType(.strikethrough)
+        case .blockquote:
+            return Gridicon.iconOfType(.quote)
+        case .orderedlist:
+            if layoutDirection == .leftToRight {
+                return Gridicon.iconOfType(.listOrdered)
+            } else {
+                return Gridicon.iconOfType(.listOrderedRTL)
+            }
+        case .unorderedlist:
+            return Gridicon.iconOfType(.listUnordered).imageFlippedForRightToLeftLayoutDirection()
+        case .link:
+            return Gridicon.iconOfType(.link)
+        case .horizontalruler:
+            return Gridicon.iconOfType(.minusSmall)
+        case .sourcecode:
+            return Gridicon.iconOfType(.code)
+        case .more:
+            return Gridicon.iconOfType(.readMore)
+        case .header1:
+            return Gridicon.iconOfType(.headingH1)
+        case .header2:
+            return Gridicon.iconOfType(.headingH2)
+        case .header3:
+            return Gridicon.iconOfType(.headingH3)
+        case .header4:
+            return Gridicon.iconOfType(.headingH4)
+        case .header5:
+            return Gridicon.iconOfType(.headingH5)
+        case .header6:
+            return Gridicon.iconOfType(.headingH6)
+        case .code:
+            return Gridicon.iconOfType(.posts)
+        default:
+            return Gridicon.iconOfType(.help)
+        }
+    }
+
+    private var layoutDirection: UIUserInterfaceLayoutDirection {
+        return UIView.userInterfaceLayoutDirection(for: .unspecified)
+    }
+
+    var accessibilityLabel: String {
```

Comment:
It might make sense to also add `accessibilityTraits` and `accessibilityHint`, to improve accessibility like in the rest of the app.

Reply:
thanks for raising the accessibility issues! I believe `accessibilityTraits` is set in a `UIButton` by default (confirmed in the view debugger). I added `accessibilityHint` for each action in https://github.com/woocommerce/woocommerce-ios/pull/1440/commits/791a8cb29eddcd93a81d1b1d64351f20117787af - the copy is from myself, feel free to suggest any edits!

I tested with VoiceOver in the editor screen, and when tapping a format bar action like Italic, now it reads "Italic, button" - pause - "Tap to toggle Italic text"

Reply:
Sounds good!